File: telegram-bot-core/src/main/java/ru/tggc/telegrambotcore/registry/resolver/UserInfoResolver.java
package ru.tggc.telegrambotcore.registry.resolver;

import org.springframework.stereotype.Component;
import ru.tggc.telegrambotcore.annotation.params.UserInfo;
import ru.tggc.telegrambotcore.dto.UserDto;

import java.lang.reflect.Parameter;

@Component
public class UserInfoResolver implements ParameterResolver<UserDto> {

    @Override
    public boolean supports(Parameter p) {
        return p.isAnnotationPresent(UserInfo.class) && p.getType().equals(UserDto.class);
    }

    @Override
    public UserDto resolve(Parameter parameter, HandlerCtx ctx) {
        return new UserDto(ctx.from().id(), ctx.from().username());
    }
}

File: telegram-bot-core/src/main/kotlin/ru/tggc/telegrambotcore/registry/resolver/UserInfoResolver.kt
package ru.tggc.telegrambotcore.registry.resolver

import org.springframework.stereotype.Component
import ru.tggc.telegrambotcore.annotation.params.UserInfo
import ru.tggc.telegrambotcore.dto.UserDto
import java.lang.reflect.Parameter

@Component
class UserInfoResolver : ParameterResolver<UserDto?> {
    override fun supports(p: Parameter?): Boolean? =
        p?.isAnnotationPresent(UserInfo::class.java) == true && p.getType() == UserDto::class.java

    override fun resolve(parameter: Parameter?, ctx: HandlerCtx): UserDto =
        UserDto(ctx.from!!.id(), ctx.from.username())

}
